src/Controller/ViewerController.php

```diff
@@ -103,4 +119,57 @@ public function editor(MediaInterface $media, Request $request, $edit = FALSE) {
     return $response;
   }
 
+  /**
+   * Gets a render array for a cool viewer.
+   *
+   * @param \Drupal\media\Entity\Media $media
+   *   The media entity to view / edit.
+   * @param string $wopi_client
+   *   The WOPI client url.
+   * @param bool $can_write
+   *   Whether this is a viewer (false) or an edit (true). Permissions will
+   *   also be checked.
+   * @param array{closebutton: bool} $options
+   *   Options for the renderer. Current values:
+   *     - "closebutton" if "true" will add a close box. (see COOL SDK)
+   *
+   * @return array|array{error: string}
+   *   A stub render element array, or an array with an error on failure.
+   *
+   * @throws \Drupal\collabora_online\Exception\CollaboraNotAvailableException
+   *   The key to use by Collabora is empty or not configured.
+   */
+  protected function getViewerRender(MediaInterface $media, string $wopi_client, bool $can_write, $options = NULL) {
```

Comment:
Actually, in the current code there is so much wrong with these `$options`:
- The value is always the same (it is only called once).
- The value is `['closebutton' => 'true']`, using string `'true'` instead of boolean `true`.
- The array only has this one key.

So for now the simplest is to drop this parameter altogether.

Reply:
Indeed